### PR TITLE
improve Hero movements and fix camera updates

### PR DESCRIPTION
**Pull request by avdonin**
The following changes is presented:

- make **self.next_step** hidden in Hero class
- added diagonal moves when two keys is pressed
- fixed the camera shaking effect while making diagonal moves
- added moves along walls when two keys is pressed
- added acceleration of the Hero

elif in _square.py_ **update**() doesn't allow to make diagonal move
self.next_step could be a list, if it is hidden
elif in _camera.py_ **update**() makes two updates while making diagonal moves
conditions before **self.make_step**() in _square.py_ will allow to move along walls if a diagonal move is made
changing STEP_TIME in interval [INIT_STEP_TIME, FASTEST_STEP_TIME] will create acceleration effect 

Potential problems:
- ability to collect eggs
- diagonal moves through obstacles

### CODE REVIEW

File: square.py [hunk 1]
```diff
@@ -19,28 +19,41 @@ def draw(self, surface, camera):
 class Hero(Square):
     ''' The Hero is a creature which is managed by the user '''
     
+    INITIAL_STEP_TIME = 0.15
+    FASTEST_STEP_TIME = 0.04
```
Comment (dvpashmentov) [inline square.py]: It will be better to declare them inside __init__(...)

**Comment by dvpashmentov** [comment]
make self.next_step hidden in Hero class: no, self._next_step isn't private. But self.__next_step is. I will do it by myself later 

added diagonal moves when two keys is pressed, 
added moves along walls when two keys is pressed: because of a bug (diagonal moves through obstacles) these ideas must be considered later (I have no time for this now)

added acceleration of the Hero: perfect (except  https://github.com/Pashmen/Find_eggs/pull/2/commits/c194b419bf76b763a29da39476983cd1962b8075#r350125412)

fixed the camera shaking effect while making diagonal moves: it's ok

Summary: 
Leave only the acceleration and the changes in Camera
https://github.com/Pashmen/Find_eggs/pull/2/commits/c194b419bf76b763a29da39476983cd1962b8075#r350125412

File: square.py (hunk 1 above)
Comment (avdonin) [reply square.py]: Alright.

File: square.py [hunk 2]
```diff
@@ -19,24 +19,25 @@ def draw(self, surface, camera):
...
+        self.INITIAL_STEP_TIME = 0.22
+        self.FASTEST_STEP_TIME = 0.04
+        self.STEP_TIME_ACCELERATION = 0.03
+
         Square.__init__(self, x, y, color)
 
         self.STEP_TIME = self.INITIAL_STEP_TIME
         self.last_step_time = time.time()
-        self._next_step = None
+        self.__next_step = None
```
Comment (dvpashmentov) [inline square.py]: Эта переменная вообще используется? :) 
Лучше оставить self.next_step

File: square.py [hunk 3]
```diff
@@ -63,18 +64,20 @@ def update(self, field, eggs, cutter, snake, pseudo_cutter):
             step_is_made = eggs[4].update(self) or step_is_made
             step_is_made = pseudo_cutter.update(field, self, eggs, cutter, snake) or step_is_made
             if not step_is_made and self.next_step != None: 
-                if field.is_valid(self.next_step[0], self.next_step[1]):            
-                    self.make_step()
+                if field.is_valid(self.next_step[0], self.next_step[1]):
+                    # the next condition will not allow diffraction-like behaviour while moving diagonally
+                    if (field.is_valid(self.x, self.next_step[1]) or field.is_valid(self.next_step[0], self.y)):
```
Comment (dvpashmentov) [inline square.py]: Лучше вместо or использовать and. 
В варианте с or hero всё равно может двигаться по диагонали сквозь препятствия

File: square.py [hunk 4]
```diff
@@ -19,28 +19,41 @@ def draw(self, surface, camera):
...
+                #print(f"STEP_TIME={self.STEP_TIME} diff={time.time() - self.last_step_time} left") 
+            if is_pressed[K_RIGHT] and self.x < field.WIDTH - 1:
+                self.next_step[0] += 1
+                #print(f"STEP_TIME={self.STEP_TIME} diff={time.time() - self.last_step_time} right")   
+            if is_pressed[K_UP] and self.y > 0:
+                self.next_step[1] -= 1
+                #print(f"STEP_TIME={self.STEP_TIME} diff={time.time() - self.last_step_time} up")
+            if is_pressed[K_DOWN] and self.y < field.HEIGHT - 1:
+                self.next_step[1] += 1
+                #print(f"STEP_TIME={self.STEP_TIME} diff={time.time() - self.last_step_time} down")
```
Comment (dvpashmentov) [inline square.py]: Эти комментарии лучше удалить

File: square.py (hunk 3 above)
Comment (dvpashmentov) [reply square.py]: Хотя не совсем так :) Простой заменой не получится обойтись. 
В общем, хотелось бы, чтобы движение по диагонали было физично. То есть, например, если есть препятствие слева, то пойти по диагонали влево вверх не получится.

Comment (dvpashmentov) [reply square.py]: Sorry for Russian. I forgot

**Review by dvpashmentov — COMMENTED** [review]
There can be a situation when the hero is moving diagonally and is moving a gray box. This box can move through objects. It isn't physical behaviour

**Review by dvpashmentov — COMMENTED** [review]
Again:) There can be a situation when the hero is moving diagonally and is moving an egg[2]. This egg can move through objects. It isn't physical behaviour

That's a last correction, I think

**Pull request merged**